extensions/coap/protocols/CoapC2Protocol.cpp

```diff
@@ -39,7 +39,7 @@ CoapProtocol::CoapProtocol(const std::string &name, const utils::Identifier &uui
 
 CoapProtocol::~CoapProtocol() = default;
 
-void CoapProtocol::initialize(const std::shared_ptr<core::controller::ControllerServiceProvider> &controller, const std::shared_ptr<Configure> &configure) {
+void CoapProtocol::initialize(core::controller::ControllerServiceProvider* controller, const std::shared_ptr<Configure> &configure) {
```

Comment:
if we take a pointer, it's our responsibility to null-check it, we assume here that it is non-null, we should either handle nullptr or take a reference, (in theory we could use a gls::non_null<...>, but references are more succinctly command validity)

Reply:
I take it back, since we intend to store the pointer (in `initialize`) I agree with the pointer